### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Loris99**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "@testing-library/jest-dom": "^5.16.2",
     "@testing-library/react": "^12.1.4",
     "@testing-library/user-event": "^13.5.0",
+    "prop-types": "^15.8.1",
```
Comment (ehabAmriahExalt) [inline package.json]: I see it instaled but not used

File: src/Components/DotLine.module.css [hunk 2]
```diff
@@ -0,0 +1,20 @@
+.contains {
+  height: 30px;
+  width: 30px;
+  border-radius: 50%;
+  margin: 5px;
+  margin-top: 10px;
+  background-color: white;
+  border: 2px solid black;
+  display: inline-block;
+}
```
Comment (ehabAmriahExalt) [inline src/Components/DotLine.module.css:10]: please add space between the classes

File: src/Components/Game.module.css [hunk 3]
```diff
@@ -0,0 +1,53 @@
+.container {
+  display: grid;
+  height: 700px;
+  /* width: 900px; */
```
Comment (ehabAmriahExalt) [inline src/Components/Game.module.css]: please remove if not needed

File: src/Components/Game.module.css [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+}
+.sideBar {
+  display: grid;
+  width: 100%;
+}
+
+.linesContainer {
+  display: grid;
+  overflow: auto;
+  /* width: 100%; */
```
Comment (ehabAmriahExalt) [inline src/Components/Game.module.css]: please remove if not needed

File: src/Components/Line.js [hunk 5]
```diff
@@ -1,47 +1,99 @@
...
 
-  // const [enteredValue, setEnteredValue] = useState([0, 0, 0, 0]);
-  const [enteredValue, setEnteredValue] = useState(["", "", "", ""]);
+  useEffect(() => {});
+  //check if input is number & add into the array
   const valueInputChangeHandler = (digitValue, index) => {
-    const clonedValues = [...enteredValue];
-    clonedValues[index] = digitValue;
+    if (DIGITS.test(digitValue) === false) {
+      console.log(DIGITS.test(digitValue));
```
Comment (ehabAmriahExalt) [inline src/Components/Line.js]: please remove console.log

File: src/Components/Game.js [hunk 6]
```diff
@@ -1,38 +1,85 @@
...
-    <div className="container">
-      <div className="top-bar">
+    <div className={styles.container}>
+      <div className={styles.topBar}>
         <h1>Find The Secret Code</h1>
       </div>
-      <div className="game-bar">
-        <Line secretCode={codeValue} />
+      <div className={styles.linesContainer}>
+        {Array(8)
```
Comment (ehabAmriahExalt) [inline src/Components/Game.js]: please move 8 to const

File: src/Components/Line.js [hunk 7]
```diff
@@ -1,47 +1,99 @@
...
+    }
+    //sort boolean array
+    arr.sort((value) => {
+      return value ? -1 : 1;
+    });
+    setCircles(arr);
+
+    if (arr.every((val) => val !== false) && arr.length === 4) {
+      props.updateIsAWin(true);
+    } else if (props.activeStep === 7) {
```
Comment (ehabAmriahExalt) [inline src/Components/Line.js]: use the const that you defined in the Game component (gameSize)

File: src/Components/Line.module.css [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+.boardAlignment {
+  margin: 5px 10px;
+  display: flex;
+  flex-direction: row;
+}
+
+.dotAlignment {
+  /* display: inline-block; */
+/* flex-basis: 40%; */
+  /* justify-content: space-between; */
```
Comment (ehabAmriahExalt) [inline src/Components/Line.module.css]: please remove if not needed

File: src/Components/Line.module.css [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+  height: 30px;
+  border-radius: 40%;
+  background-color: rgb(0, 150, 219);
+  border-color: rgb(122, 200, 236);
+  text-align: center;
+  text-decoration: none;
+  font-size: 14px;
+  color: black;
+  margin: 10px;
+  /* flex-basis: 10%; */
```
Comment (ehabAmriahExalt) [inline src/Components/Line.module.css]: please remove if not needed

File: src/Components/Line.module.css [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+  flex-direction: row;
+}
+
+.dotAlignment {
+  /* display: inline-block; */
+/* flex-basis: 40%; */
+  /* justify-content: space-between; */
+  align-content: flex-start;
+  margin-right: 10px;
+  /* margin-top: 10px; */
```
Comment (ehabAmriahExalt) [inline src/Components/Line.module.css]: please remove if not needed

**Review by ehabAmriahExalt — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Loris99) [reply package.json]: done

Comment (Loris99) [reply package.json]: removed it

File: src/Components/DotLine.module.css (hunk 2 above)
Comment (Loris99) [reply src/Components/DotLine.module.css:10]: done

File: src/Components/Game.module.css (hunk 3 above)
Comment (Loris99) [reply src/Components/Game.module.css]: done

File: src/Components/Game.module.css (hunk 4 above)
Comment (Loris99) [reply src/Components/Game.module.css]: done

File: src/Components/Line.js (hunk 5 above)
Comment (Loris99) [reply src/Components/Line.js]: done

File: src/Components/Game.js (hunk 6 above)
Comment (Loris99) [reply src/Components/Game.js]: done

File: src/Components/Line.module.css (hunk 8 above)
Comment (Loris99) [reply src/Components/Line.module.css]: done

File: src/Components/Line.module.css (hunk 9 above)
Comment (Loris99) [reply src/Components/Line.module.css]: done

File: src/Components/Line.module.css (hunk 10 above)
Comment (Loris99) [reply src/Components/Line.module.css]: done

File: src/Components/Line.js (hunk 7 above)
Comment (Loris99) [reply src/Components/Line.js]: done

**Review by ehabAmriahExalt — APPROVED** [review]

**Pull request merged**